### ISSUE TITLE
Search does not return results in currently selected version

### ISSUE THREAD

**Issue by rmloveland** [cockroachdb/docs#11056]
Richard Loveland (rmloveland) commented:

We are getting feedback that folks are having a hard time finding our docs on the [Duplicate indexes multi-region pattern](https://www.cockroachlabs.com/docs/v20.2/topology-duplicate-indexes), and some other things.

We recently removed those pages from the v21.1 docs, but they still exist in the v20.2 and earlier docs (see link above).

Current behavior:

If you are visiting [a page in the v20.2 docs](https://www.cockroachlabs.com/docs/v20.2/topology-duplicate-indexes) and perform a search from the left hand nav search bar for "duplicate index", the [search results](https://www.cockroachlabs.com/docs/search.html?query=duplicate+index) do not contain any of the v20.2 pages that contain the term "duplicate index".

![Screen Shot 2021-08-16 at 3 09 44 PM](https://user-images.githubusercontent.com/1312291/129616893-9144ea86-b646-45fd-b92f-8855a387db40.png)

In other words, as a user I expect that if I am browsing a specific version of CRDB docs, that the search box will give me results from the currently selected version in the version dropdown.

Expected behavior: 

If I have selected v20.2 from the page version dropdown, then a search from the left nav search bar will return results from the v20.2 docs.

Jira Issue: DOC-1475

**Comment by awoods187** [comment]
@timveil mentioned to me that this is a big problem for customers and the field. Do we need to seek out help to address this limitation? cc @lnhsingh

**Comment by lnhsingh** [comment]
Some background: When we did the search revamp (end of 2019 / beginning of 2020), we decided to show search results from the stable and dev versions of the docs. IIRC, this was to cut down on the number of repeated results due to many pages in different versions being basically identical. The search isn't smart enough (right now, at least) to know what version the user is currently on and then to search that specific version of the docs. Since it's only searching stable / dev, v20.2 results won't be returned, now that we have v21.2 as dev.

Search results from the supported releases would be ideal, but I'm not sure if it should be filtered based on the version you're on, or if the search should just return all results across all supported versions. We'd probably want some input from Design (cc @luciaaaaaaa).

We would probably want to see if @anagio is available to help with this since YZ left. Otherwise, we might want to see if DevOps (?) can help.

**Comment by luciaaaaaaa** [comment]
I think it would be nice to explore what the lift would be to implement what Rich outlines: If I have selected v20.2 from the page version dropdown, then a search from the left nav search bar will return results from the v20.2 docs.

**Comment by lnhsingh** [comment]
Ok, I can see if we can get an engineering resource. The quick, intermediary fix might to add v20.1 and v20.2 docs to the index so all supported versions are searchable

**Comment by awoods187** [comment]
Is there an update on this issue?

**Comment by lnhsingh** [comment]
Either solution (searching all versions or searching the version the user is currently on) would require us to index more versions of the pages (we currently only index stable and dev versions). @anagio reached out to Algolia to see if they had any suggestions on how to handle this kind of related data / old versions (without bloating the number of records we have) and they just pointed him to their [docs](https://docsearch.algolia.com/docs/required-configuration/#introduce-global-information-as-meta-tags), so it seems we would have to index all of the content.

That being said, this is a more involved update than we currently have the bandwidth for. It's a priority to hire a Product Ops Eng, and this update can be a project they work on when they join. The current search behavior is doing what we intended it to do (search stable / dev versions only), so unless we get more feedback that this is not working for users, we will have to wait until we have the engineering resources to work on this.

Keeping this issue open -- please add to it if you receive more feedback from customers / the field. Thank you!

**Comment by awoods187** [comment]
Thanks Lauren. We've mitigated this through the new migration page but will leave this issue open in case we have additional options in the future.
